client/js/main.ts

```diff
@@ -68,57 +109,17 @@ function readURLHash() {
 readURLHash();
 window.addEventListener("hashchange", readURLHash);
 
-interface IGraphqlTag {
-	tag: {
-		name: string
-	};
-	checked_in: boolean;
-	checked_in_by?: string;
-	checked_in_date?: string;
-}
-
-interface IGraphqlQuestion {
-	name: string;
-	value: string;
-}
-
-interface IGraphqlAttendee {
-	user: {
-		id: string,
-		name: string,
-		email: string,
-		questions?: IGraphqlQuestion[]
-	};
-	tags: IGraphqlTag[];
-}
-
-interface ISearchUserResponse {
-	data: {
-		search_user_simple: IGraphqlAttendee[];
-	}
-}
-
-interface ITagChangeResponse {
-	tag_change: IGraphqlAttendee;
-}
-
-const graphqlOptions = {
-	dataType: "text",
-	responseType: "json",
-	headers: {
-		"Content-Type": "application/json",
-		"Accept": "application/json"
-	}
-}
-
 function delay (milliseconds: number) {
 	return new Promise<void>(resolve => {
 		setTimeout(resolve, milliseconds);
 	});
 }
 
-function statusFormatter (time: string, by: string = "unknown"): string {
+function statusFormatter (time: string, by: string | null): string {
 	// Escape possible HTML in username
+	if (by === null) {
```

Comment:
Any reason to do this instead of an optional argument?

Reply:
I originally did this because the type of `checked_in_by` in the GraphQL Tag object is `string | null`. But GraphQL queries should always return a string for `checked_in_by` (see https://github.com/HackGT/checkin2/blob/master/server/graphql.ts#L129), so I just went ahead and made this a non-null type.